### DISCUSSION TITLE
[Question] How can I selectively enforce HTTPS on a health check endpoint

### DISCUSSION THREAD

**Discussion by Bo-Simmons** [dotnet/aspnetcore#25374] [General]
I've configured [health checks](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-3.1) using the middleware in an MVC application, but have been asked to enforce https on the health endpoint for security reasons. For separate reasons, I can't enforce it across the service as a whole, and so I was wondering if there was a mechanism to selectively do this to just the health endpoint. While it's possible to do this with a controller, but I can't seem to achieve it for the endpoint created by the middleware.

Is there any built-in way to achieve this? Could I create some sort of custom middleware as a wrapper to achieve it?

**Comment by Tratcher** [comment]
Similar recent discussions:
https://github.com/dotnet/aspnetcore/issues/25221
https://github.com/dotnet/aspnetcore/issues/21506
